crates/sui-indexer/src/restorer/formal_snapshot.rs

```diff
@@ -0,0 +1,231 @@
+                                                    IndexedDeletedObject {
+                                                        object_id: obj_key.0,
+                                                        object_version: obj_key.1.value(),
+                                                        checkpoint_sequence_number: self.next_checkpoint_after_epoch,
+                                                    },
+                                                );
+                                            }
+                                        }
+                                    }
+                                });
+                            let live_obj_cnt = move_objects.len();
+                            let wrapped_or_deleted_obj_cnt = wrapped_or_deleted_move_objects.len();
+                            let object_changes = TransactionObjectChangesToCommit {
+                                changed_objects: move_objects.clone(),
+                                deleted_objects: vec![],
+                            };
+                            info!("Start persisting {} objects to objects table from {}", live_obj_cnt, object_file_path);
+                            store_clone
+                                .persist_objects(vec![object_changes])
+                                .await
+                                .expect("Failed to persist to objects from restore");
+                            info!("Finished persisting {} objects to objects table from {}", live_obj_cnt, object_file_path);
+                            info!("Start persisting {} objects and {} wrapped or deleted objects to objects snapshot from {}", live_obj_cnt, wrapped_or_deleted_obj_cnt, object_file_path);
+                            let objects_snapshot_changes = TransactionObjectChangesToCommit {
+                                changed_objects: move_objects.clone(),
+                                deleted_objects: wrapped_or_deleted_move_objects,
+                            };
+                            info!("Finished persisting {} objects and {} wrapped or deleted objects to objects snapshot from {}", live_obj_cnt, wrapped_or_deleted_obj_cnt, object_file_path);
+
+                            store_clone
+                                .persist_objects_snapshot(vec![objects_snapshot_changes])
+                                .await
+                                .expect("Failed to persist objects snapshot");
+                            bar_clone.inc(1);
+                            bar_clone.set_message(format!("Restored {} live move objects and {} wrapped or deleted objects from {}", live_obj_cnt, wrapped_or_deleted_obj_cnt, object_file_path));
+                            Ok::<(), anyhow::Error>(())
+                        }
+                    })
+                    .boxed()
+                    .buffer_unordered(self.restore_config.object_store_concurrent_limit)
```

Comment:
While this does do things concurrently, its not done in a parallelizable way. we may want to spawn tasks so that we can make use of more cores of the machine.